### PR TITLE
FS4004 Optimise fund dashboard endpoint

### PR DESCRIPTION
**Pull request by ksp37-dluhc**
### Change description
The fund dashboard endpoint makes several calls to store services. This PR collates calls where possible and executes them concurrently.

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Page load times on Test

### CODE REVIEW

File: app/blueprints/assessments/routes.py [hunk 1]
```diff
@@ -134,6 +137,21 @@
 )
 
 
+def run_with_context(func, *args, **kwargs):
+    with current_app.app_context():
+        return func(*args, **kwargs)
+
+
+class ContextThreadPoolExecutor(concurrent.futures.ThreadPoolExecutor):
```
Comment (nuwan-samarasinghe) [inline app/blueprints/assessments/routes.py:145]: I think in utils we have this ContextThreadPoolExecutor can you check is it can be used

**Review by nuwan-samarasinghe — DISMISSED** [review]

File: app/blueprints/assessments/routes.py (hunk 1 above)
Comment (ksp37-dluhc) [reply app/blueprints/assessments/routes.py:145]: I searched for it but I couldn't find anything?

Comment (nuwan-samarasinghe) [reply app/blueprints/assessments/routes.py:145]: may be check this https://github.com/communitiesuk/funding-service-design-utils/blob/main/fsd_utils/sqs_scheduler/context_aware_executor.py

Comment (ksp37-dluhc) [reply app/blueprints/assessments/routes.py:146]: Ooo nice - I've updated to use that instead 👍🏽

**Review by hamzabinkhalid — APPROVED** [review]

**Pull request merged**